### PR TITLE
Control caching based on response

### PR DESCRIPTION
**Pull request by volodymyrss**
I might be missing entierly the intent of the developers, and possibly there is a much better solution to my problem (#91). 
This is just what I implemented which is sufficient for my needs, I wonder if it makes any sense.

### CODE REVIEW

**Comment by sh4nks** [comment]
This looks good to me from a first look! I'll take a closer look when I am home from work.

File: tests/test_view.py [hunk 1]
```diff
@@ -55,6 +55,41 @@ def cached_view():
 
     assert the_time == rv.data.decode('utf-8')
 
+def test_cached_view_response_filter(app, cache):
+
+    @app.route('/a')
+    @cache.cached(5, response_filter=lambda x: x[1]<400)
+    def cached_view():
+        print("app.rt",app.return_code)
```
Comment (sh4nks) [inline tests/test_view.py]:
```suggestion
```
Please delete the print statement

**Comment by volodymyrss** [comment]
thanks for the suggestion!
I see you do not use print even in the test, I should have followed that. Could I ask why, for educational purposes?

**Comment by sh4nks** [comment]
Wait, you were able to commit my suggested changes directly from github? 😄 

Generally speaking, instead of using print statements, the information should be logged using a dedicated logger. The logger can then be configured to only print certain log levels to the console, or log everything to a file.

In our case, by reading the test cases you have written one can easily guess what the result is -- and in case the tests fail, pytest will show us all the needed information anyway.

Hope that clears things up!

**Comment by volodymyrss** [comment]
Ok, I see, thanks!
I would usually use logger except for the tests, but then no reason to treat tests as second-class code. 

Yeah, the technology of github is growing beyond our imagination. There was just a button to accept the suggestion, first time I see such a feature.

**Pull request merged**